src/main/java/org/ja13/eau/sixnode/electriccable/ElectricCable.kt

```diff
@@ -201,11 +209,36 @@ class PlayerHarmer(val electricalLoad: org.ja13.eau.sim.ElectricalLoad, private
         val objects = location.world().getEntitiesWithinAABB(Entity::class.java, location.getAxisAlignedBB(4))
         for(obj in objects) {
             val ent = obj as Entity
+            if(ent is EntityLiving || ent is EntityPlayerMP){
             val distance = location.distanceTo(ent)
             val pain = (harmFunction(distance) * harmLevel).toFloat()
-            if (distance < 3 && pain > 0.05) {
+            if(distance < 1 && pain > 0.05 && ent.onGround){ //if they're touching the block, deal extra effects
+                if(ent is EntityCreeper){
+                    location.world().createExplosion(ent, ent.posX, ent.posY, ent.posZ, 3f, true)
+                    ent.customNameTag = "Electrically-detonated Creeper"
+                    ent.setDead()
+                }
+                if(ent is EntityEnderman){ ent.setFire(1) } //enderman will teleport
+                ent.hurtResistantTime = 0
                 ent.attackEntityFrom(DamageSource("Cable"), pain)
+                if(ent.isEntityAlive) {
+                    ent.setVelocity(Math.min(location.directionOf(ent).x*pain,1.0), Math.min(0.5*pain,3.0), Math.min((location.directionOf(ent).z*pain),1.0))
+                }
             }
+            else if (distance < 3 && pain > 0.05 && ent.hurtResistantTime == 0 && ent.onGround) { //otherwise deal less punishing effects
+                ent.attackEntityFrom(DamageSource("Cable"), pain)
+                ent.hurtResistantTime = 15
+                if(ent is EntityCreeper){
+                    location.world().createExplosion(ent, ent.posX, ent.posY, ent.posZ, 3f, true)
+                    ent.customNameTag = "Electrically-detonated Creeper"
+                    ent.setDead()
+                }
+                if(ent is EntityEnderman){ ent.setFire(1) } //enderman will teleport
+                if(ent.isEntityAlive){
+                    ent.setVelocity(0.0,Math.min(pain/2.0,0.65),0.0) //Electricity makes you jump
+                }
+            }
+        }
```

Comment:
The two sides of the if are a lot of duplicated code, I'd rather see it as one entry for code and to change the values where applicable based on the values. It will make this code execute faster, which is kinda important since it happens per tick for a lot of stuff.

Reply:
I'll go ahead and optimize that.

Reply:
Thank you very much for bringing this up. The new codebase is MUCH cleaner and i'm glad I put the energy into fixing it.